korge/src/commonMain/kotlin/com/soywiz/korge/render/BatchBuilder2D.kt

```diff
@@ -662,15 +662,61 @@ class TexturedVertexArray(var vcount: Int, val indices: IntArray, var isize: Int
     /** Sets the [cMul] and [cAdd] (multiplicative and additive colors) of the vertex previously selected calling [select] */
 	fun cols(colMul: RGBA, colAdd: Int) = setCMul(colMul).setCAdd(colAdd)
 
+    private fun quadV(pos: Int, x: Float, y: Float, u: Float, v: Float, colMul: RGBA, colAdd: Int) {
+        f32[pos + 0] = x
+        f32[pos + 1] = y
+        f32[pos + 2] = u
+        f32[pos + 3] = v
+        i32[pos + 4] = colMul.value
+        i32[pos + 5] = colAdd
+    }
+
     /**
      * Sets a textured quad at vertice [index] with the region defined by [x],[y] [width]x[height] and the [matrix],
      * using the texture coords defined by [BmpSlice] and color transforms [colMul] and [colAdd]
      */
+    @OptIn(KorgeInternal::class)
 	fun quad(index: Int, x: Double, y: Double, width: Double, height: Double, matrix: Matrix, bmp: BmpSlice, colMul: RGBA, colAdd: Int) {
```

Comment:
In KorGE 2.0 we should consider using Float for Matrix, or creating a Matrixf using floats